File: wag/static/model/dataObjects/features.js
//** Object Feature **//
	
	var features = new Object();
	
/**
 * Creates an instance of an accessibility feature.
 *
 * @constructor
 * @this {feature}
 * @param {Object} data The xml object that holds data of the feature.
 */		
	var feature = function(data){  
		this.title = $(data).find("TITLE").text();
		this.disability = $(data).find("DISABILITY").text();
		this.rank = $(data).find("DISABILITY").attr('rank');
		this.priority = $(data).find("PRIORITY").text();
		this.level = $(data).find("PRIORITY").attr('level');
		this.error = $(data).find("ERROR").text();
		this.description = $(data).find("DESCRIPTION").text(); 
		this.solution = $(data).find("SOLUTION").text();
		
		this.rule = eval(rules[$(data).attr('id')]); 
	};  	
	
/**
 * Creates the collection of new feature objects.
 *
 * @constructor
 */	
	function makeFeatureCollection(xml) { 
		$(xml).find('FEATURE').each(function() {
			features[$(this).attr('id')] = new feature(this);
		}); 
	}
	

	

File: wag/core/Features.py
from bs4 import BeautifulSoup
#xml =open("web_accessibility_spec.xml")
#soup = BeautifulSoup(xml, "xml")

class Feature:
    def __init__(self, title, disability,rank,priority,level,error,description,solution,rule):
        self.title = title
        self.disability = disability
        self.rank = rank
        self.priority = priority
        self.level = level
        self.error = error
        self.description = description
        self.solution = solution
        self.rule = rule 
        
    
class Features:      
    def makeFeatureCollection (self):
        features = [];
        for i in soup.findAll("FEATURE"):
            features.append(Feature(i.TITLE,i.DISABILITY,i.DISABILITY['rank'],i.PRIORITY,i.PRIORITY['level'],i.ERROR.get_text(),i.DESCRIPTION,i.SOLUTION,i['id']))
        return features